### PR TITLE
Replace referee with @sinonjs/referee

### PR DESCRIPTION
**Pull request by mroderick**
This PR replaces the `referee` dependency with `@sinonjs/referee`.

#### How to verify - mandatory

1. Observe that all tests pass on Travis
2. ?
3. Win!

### CODE REVIEW

**Comment by mantoni** [comment]
Build is failing due to a timeout. Restarted it, let's see whether it was just a one-off thing.

**Comment by mroderick** [comment]
It happens somewhat frequently in this project, re-starting the tests fixes it. I haven't had time to look into it yet

**Comment by mantoni** [comment]
Yes, not it passed. Maybe we should just increase the timeout then.

File: package.json [hunk 1]
```diff
@@ -24,7 +24,6 @@
     "test": "mocha 'lib/**/*.test.js'",
     "test:coverage": "nyc --reporter=lcov --reporter=text --all npm test -- --reporter dot",
     "precommit": "npm run lint -- --fix && npm run test",
-    "prepublishOnly": "mkdocs gh-deploy -r upstream || mkdocs gh-deploy -r origin",
```
Comment (mantoni) [inline package.json]: You didn't mention this in the description. Is it deleted intentionally?

Comment (mantoni) [reply package.json]: Ignore me. I just saw it's a duplicate line.

**Pull request merged**